llvm/lib/Target/AArch64/AArch64FrameLowering.cpp

```diff
@@ -2139,14 +2139,16 @@ void AArch64FrameLowering::determineCalleeSaves(MachineFunction &MF,
   if (BigStack || !CanEliminateFrame || RegInfo->cannotEliminateFrame(MF))
     AFI->setHasStackFrame(true);
 
+  bool ShouldSpillScavengeSlot = MF.getTarget().Options.MCOptions.RegisterScavengingSpillSlot;
+
   // Estimate if we might need to scavenge a register at some point in order
   // to materialize a stack offset. If so, either spill one additional
   // callee-saved register or reserve a special spill slot to facilitate
   // register scavenging. If we already spilled an extra callee-saved register
   // above to keep the number of spills even, we don't need to do anything else
   // here.
-  if (BigStack) {
-    if (!ExtraCSSpill && UnspilledCSGPR != AArch64::NoRegister) {
+  if (BigStack || ShouldSpillScavengeSlot) {
```

Comment:
These two conditionals are especially confusing.
I know this is that minimal change without having to shift things too much around, but it took me a solid minute to understand what is happening, even though I had this patch and the discussion in [D33443][1] in working memory.

I'd suggest the following minor changes:

1) Rename the option to something like `ForceSpillScavengingSlot` or similar to denote that this will happen no matter what; "should" has an optional or permissive quality to it.
2) Add a comment explaining that we want to avoid this handling when the cmd line option is set and fall into the next case that creates an emergency spill slot on the stack.

[1]: https://reviews.llvm.org/D33443

Reply:
That makes sense! Will do.

Reply:
@compor Let me know if this is better now!

Reply:
LGTM